### PR TITLE
Update ES Lint and editor config

### PR DESCRIPTION
**Pull request by joeyfigaro**
- Updated eslint config 
- Added editorconfig for consistency across contributions and updated formatting on source files
- Added lint script to package.json
- Fixed linter issues with source files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "prepublish": "npm test && npm run build",
     "start": "node examples/server.js",
+		"lint": "eslint src/*",
```
Comment (pburtchaell) [inline package.json]: Can you fix the indent here?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# top-most EditorConfig file
+root = true
+
+[*.js]
+indent_style = space
+indent_size = 4
```
Comment (pburtchaell) [inline .editorconfig]: The indent size should be `2` 😄

File: src/defaultPropTypes.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+        PropTypes.node,
+    ]),
+    onClick: PropTypes.func,
+    style: PropTypes.bool,
+    actionStyle: PropTypes.object,
+    barStyle: PropTypes.object,
+    activeBarStyle: PropTypes.object,
+    dismissAfter: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.number,
```
Comment (pburtchaell) [inline src/defaultPropTypes.js]: This should be two spaces.

File: src/notification.js [hunk 4]
```diff
@@ -2,165 +2,168 @@ import React, { Component } from 'react';
 import defaultPropTypes from './defaultPropTypes';
 
 class Notification extends Component {
```
Comment (pburtchaell) [inline src/notification.js]: It looks like everything in this class was already two spaces. I don't think you need this diff.

File: src/notificationStack.js [hunk 5]
```diff
@@ -1,60 +1,55 @@
...
 import StackedNotification from './stackedNotification';
 
 function defaultStyleFactory(index, style) {
-  return Object.assign(
-    {},
-    style,
-    { bottom: `${2 + index * 4}rem` }
-  );
+    return Object.assign(
+        {},
```
Comment (pburtchaell) [inline src/notificationStack.js]: This should be two spaces.

**Comment by pburtchaell** [comment]
@joeyfigaro Thanks for the PR! I left some comments in my review.

**Comment by joeyfigaro** [comment]
@pburtchaell Thanks man - I'll have these changes taken care of today.

**Comment by joeyfigaro** [comment]
@pburtchaell Changes up for ya! Check out the tests. I ended up disabling eslint on a couple lines due to its complaints.

**Comment by pburtchaell** [comment]
Thanks, @joeyfigaro. I'll check this out when I have some time this week.

File: src/defaultPropTypes.js [hunk 6]
```diff
@@ -17,11 +17,11 @@ export default {
   activeBarStyle: PropTypes.object,
   dismissAfter: PropTypes.oneOfType([
     PropTypes.bool,
-    PropTypes.number
+    PropTypes.number,
```
Comment (BerkeleyTrue) [inline src/defaultPropTypes.js]: @pburtchaell Do you want dangling commas?

@joeyfigaro This seems to be an unnecessary addition.

File: src/index.js [hunk 7]
```diff
@@ -1,2 +1,2 @@
-export Notification from './notification';
-export NotificationStack from './notificationStack';
+export { Notification } from './notification';
+export { NotificationStack } from './notificationStack';
```
Comment (BerkeleyTrue) [inline src/index.js]: Are these semantics correct? Note that both Notification and NotificationStack are default exports and not named exports.

I believe the correct syntax is:

```js
export { default as Notification } from './notification';
```

https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=true&presets=es2015%2Creact&code=export%20%7B%20default%20as%20a%20%7D%20from%20'.%2Fa'%3B%0Aexport%20%7B%20a%20%7D%20from%20'.%2Fa'%3B

File: src/notification.js [hunk 8]
```diff
@@ -66,19 +67,19 @@ class Notification extends Component {
...
       OTransform: 'translatez(0)',
-      transform: 'translatez(0)'
+      transform: 'translatez(0)',
     };
 
-    return isActive ? Object.assign({}, baseStyle, {
-      left: '1rem'
-    }, barStyle, activeBarStyle) : Object.assign({}, baseStyle, barStyle);
+    return isActive ?
+    Object.assign({}, baseStyle, { left: '1rem' }, barStyle, activeBarStyle) :
```
Comment (BerkeleyTrue) [inline src/notification.js]: Yes, this is much cleaner.

File: src/notification.js [hunk 9]
```diff
@@ -89,67 +90,69 @@ class Notification extends Component {
...
-              ref="action"
-              className="notification-bar-action"
-              onClick={this.handleClick}
-              style={this.getActionStyle()}
-            >
-              {this.props.action}
-            </span>
-          ) : null}
-        </div>
+      <div className="notification-bar-wrapper">
```
Comment (BerkeleyTrue) [inline src/notification.js]: This `div` is a child of the `div` above and should be indent once to show that.

File: src/notification.js [hunk 10]
```diff
@@ -89,67 +90,69 @@ class Notification extends Component {
...
+      {this.props.title ? (
```
Comment (BerkeleyTrue) [inline src/notification.js]: Same here, children should be indented once from their parents

File: src/notification.js [hunk 11]
```diff
@@ -89,67 +90,69 @@ class Notification extends Component {
...
+          style={this.getTitleStyle()}
+        >
+        {this.props.title}
+        </span>
+      ) : null}
+      <span
+        ref="message"
+        className="notification-bar-message"
+      >
+      {this.props.message}
```
Comment (BerkeleyTrue) [inline src/notification.js]: Same here

File: src/defaultPropTypes.js (hunk 6 above)
Comment (pburtchaell) [inline src/defaultPropTypes.js]: No, we should not have dangling commas. Thanks for catching this, @BerkeleyTrue.

Comment (joeyfigaro) [inline src/defaultPropTypes.js]: @pburtchaell seconded - we should sweep through our lint configuration or decide on a standard?

File: src/index.js (hunk 7 above)
Comment (joeyfigaro) [inline src/index.js]: `export { Notification as default } from './notification';` <- default is flipped onto the other side.

File: src/defaultPropTypes.js (hunk 6 above)
Comment (joeyfigaro) [inline src/defaultPropTypes.js]: @BerkeleyTrue Dangling commas are part of the airbnb styleguide - they were added when I linted the project so it would pass. I prefer to go without, but what do you guys think?

File: src/index.js (hunk 7 above)
Comment (BerkeleyTrue) [inline src/index.js]: @joeyfigaro That exports the named property Notification as the default for this file.

Comment (pburtchaell) [inline src/index.js]: @joeyfigaro Can you remove this change from the PR? It's not related to the ES Lint and editor config changes and we can address this at a later time.

File: src/defaultPropTypes.js (hunk 6 above)
Comment (pburtchaell) [inline src/defaultPropTypes.js]: @joeyfigaro Let's add an exception for dangling commas. We don't need them.

**Comment by joeyfigaro** [comment]
@BerkeleyTrue @pburtchaell Changes up for ya'll.

File: .eslintrc [hunk 12]
```diff
@@ -11,6 +11,9 @@
     "modules": true
   },
   "rules": {
+    "comma-dangle": ["error", "never"],
```
Comment (BerkeleyTrue) [inline .eslintrc]: To stay consistent with the other rules "error" should be replaced with 2

File: src/index.js [hunk 13]
```diff
@@ -1,2 +1,2 @@
-export Notification from './notification';
-export NotificationStack from './notificationStack';
+export * from './notification';
+export * from './notificationStack';
```
Comment (BerkeleyTrue) [inline src/index.js]: Since both of these modules have no named exports, these two will not export anything.

Please change this to the form
`export { default as Notification } from './notification';`

**Comment by joeyfigaro** [comment]
@BerkeleyTrue How's she lookin'?

File: src/notificationStack.js [hunk 14]
```diff
@@ -11,38 +11,33 @@ function defaultStyleFactory(index, style) {
...
-          index,
-          notification.activeBarStyle
-        );
+* The notification list does not have any state, so use a
+* pure function here. It just needs to return the stacked array
+* of notification components.
+*/
+const NotificationStack = props => (
+  <div className="notification-list">
+  {props.notifications.map((notification, index) => {
```
Comment (BerkeleyTrue) [inline src/notificationStack.js]: This child needs to be indented once in from the parent.

File: test/mockNotification.js [hunk 15]
```diff
@@ -8,12 +8,12 @@ export default {
     return;
   },
   barStyle: {
-    background: 'rgb(2, 2, 2)'
+    background: 'rgb(2, 2, 2)',
```
Comment (BerkeleyTrue) [inline test/mockNotification.js]: Dangling comma

File: test/mockNotification.js [hunk 16]
```diff
@@ -8,12 +8,12 @@ export default {
...
   },
   actionStyle: {
-    color: 'rgb(2, 2, 2)'
+    color: 'rgb(2, 2, 2)',
```
Comment (BerkeleyTrue) [inline test/mockNotification.js]: Dangling comma

File: test/mockNotification.js [hunk 17]
```diff
@@ -8,12 +8,12 @@ export default {
...
   },
   activeBarStyle: {
-    left: '4rem'
-  }
+    left: '4rem',
```
Comment (BerkeleyTrue) [inline test/mockNotification.js]: Dangling comma

File: test/mockNotification.js [hunk 18]
```diff
@@ -8,12 +8,12 @@ export default {
...
+  },
```
Comment (BerkeleyTrue) [inline test/mockNotification.js]: Dangling comma

File: test/notification.js [hunk 19]
```diff
@@ -32,7 +32,8 @@ describe('<Notification />', () => {
     expect(component).to.have.className('notification-bar');
   });
 
- it('has custom class name', () => {
+  it('has custom class name', () => {
+    // eslint-disable-next-line
```
Comment (BerkeleyTrue) [inline test/notification.js]: Why are these added?

File: test/notification.js [hunk 20]
```diff
@@ -124,9 +127,10 @@ describe('<Notification />', () => {
   });
 
   it('should use custom active styles', () => {
+    // eslint-disable-next-line
     let component = shallow(
       <Notification
-        isActive={true}
+        isActive
```
Comment (BerkeleyTrue) [inline test/notification.js]: Let's keep this as it was. Explicit is better than implicit.

File: test/notificationStack.js [hunk 21]
```diff
@@ -7,7 +8,7 @@ describe('<NotificationStack />', () => {
   beforeEach(() => {
     notifications = [
       mockNotification,
-      Object.assign({}, mockNotification, { key: 2222 })
+      Object.assign({}, mockNotification, { key: 2222 }),
```
Comment (BerkeleyTrue) [inline test/notificationStack.js]: dangling comma

File: test/setup.js [hunk 22]
```diff
@@ -27,7 +27,7 @@ Object.keys(document.defaultView).forEach((property) => {
 });
 
 global.navigator = {
-  userAgent: 'node.js'
+  userAgent: 'node.js',
```
Comment (BerkeleyTrue) [inline test/setup.js]: dangling comma

**Comment by joeyfigaro** [comment]
@BerkeleyTrue Thanks! More updates 👍

**Comment by pburtchaell** [comment]
What's the status here?

**Comment by joeyfigaro** [comment]
@pburtchaell Believe all should be taken care of, except that test that was pulled over. Not entirely sure what the intention of the original author was. Can remove if you guys want.
